cdap-proto/src/main/java/co/cask/cdap/proto/security/Authorizable.java

```diff
@@ -74,14 +75,28 @@ public static Authorizable fromString(String entityString) {
     String typeString = typeAndId[0];
     EntityType type = EntityType.valueOf(typeString.toUpperCase());
     String idString = typeAndId[1];
-    List<String> idParts;
-    if (type != EntityType.KERBEROSPRINCIPAL) {
-      // kerberos principal might contain . which is also EntityId.IDSTRING_PART_SEPARATOR_PATTERN so don't split for
-      // them and also principal doesn't have other parts so just initialize
-      idParts = Arrays.asList(EntityId.IDSTRING_PART_SEPARATOR_PATTERN.split(idString));
-    } else {
-      idParts = Collections.singletonList(idString);
+
+    List<String> idParts = Collections.emptyList();
```

Comment:
I think we can use `new ArrayList()` here so that we do not need to do the initialize at Line 91?